sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala

```diff
@@ -225,10 +225,6 @@ class ParquetToSparkSchemaConverter(
           s"Invalid map type: $field")
 
         val keyType = keyValueType.getType(0)
-        ParquetSchemaConverter.checkConversionRequirement(
-          keyType.isPrimitive,
-          s"Map key type is expected to be a primitive type, but found: $keyType")
-
```

Comment:
I can track that this is from here https://github.com/apache/spark/pull/6617/files#r34599675. But currently I think Spark SQL might not have this specific requirement. cc @cloud-fan

Reply:
CC @liancheng  can you take a look?